File: gameserver/src/main/java/l2s/gameserver/model/base/RestartType.java
package l2s.gameserver.model.base;

public enum RestartType
{
	TO_VILLAGE,
	TO_CLANHALL,
	TO_CASTLE,
	TO_FORTRESS,
	TO_PORTAL,
	FIXED,
	AGATHION,
	ADVENTURES_SONG;

	public static final RestartType[] VALUES;

	static
	{
		VALUES = values();
	}
}

File: gameserver/src/main/java/l2s/gameserver/model/base/RestartType.kt
package l2s.gameserver.model.base

/**
 * @author VISTALL
 * @date 13:00/27.04.2011
 */
enum class RestartType {
    VILLAGE,
    AGIT,
    CASTLE,
    BATTLE_CAMP,
    FORTRESS,
    ORIGINAL,
    ADVENTURERS_SONG,
    TOWN; // TODO

    companion object {

        @JvmField
        val VALUES: Array<RestartType> = values()

        fun find(name: String): RestartType {
            return valueOf(name.toUpperCase())
        }

    }
}